lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -123,6 +149,25 @@ class UserProfileBloc {
     return user;
   }
 
+  void _listenUserActions() {
+    _userActionsController.stream.listen((action) {
+      var handler = _actionHandlers[action.runtimeType];
+      if (handler != null) {
+        handler(action).catchError((e) => action.resolveError(e));
+      }
+    });
+  }
+
+  Future _updateSecurityModel(UpdateSecurityModel updateSecurityModelAction) async {
+    if (updateSecurityModelAction.pinCode != null) {
+      await _secureStorage.write(key: 'pinCode', value: updateSecurityModelAction.pinCode);
```

Comment:
ideally we should only write to the secure storage if current pintCode is different than the new pinCode.

Reply:
👍